File: fitness-tracker/src/app/model/WorkoutDTOs/EditWorkoutDTO.ts
import { EditExerciseSetDTO } from '../ExerciseSetDTOs/EditExerciseSetDTO';

export interface EditWorkoutDTO {
  id: number;
  clientId?: number;
  name: string;
  comment: string;
  scheduled: string;
  sets: Array<EditExerciseSetDTO>;
  forDeleteSetIds: Array<number>;
}

File: fitness-tracker-backend/src/main/java/tech/fitnesstackerbackend/fitnesstrackerbackend/model/workout/dtos/EditWorkoutDTO.java
package tech.fitnesstackerbackend.fitnesstrackerbackend.model.workout.dtos;


import lombok.Data;
import tech.fitnesstackerbackend.fitnesstrackerbackend.model.exerciseset.EditExerciseSetDTO;

import java.util.List;

@Data
public class EditWorkoutDTO {
    private Long id;
    private Integer clientId;
    private String name;
    private String comment;
    private String scheduled;
    private List<EditExerciseSetDTO> sets;
    private List<Long> forDeleteSetIds;
}
